File: Gitbase/gitbase-pages/src/shared/services/users.service.ts
import { AuthData } from "../models/auth-data.model";
import type { AuthResponse } from "../models/responses/auth.response";
import type { User } from "../models/user.model";
import { AuthorizedService } from "./base-services/authorized.service";

export class UsersService extends AuthorizedService {
  CONTROLLER_NAME = 'users';
  CONTROLLER_URL  = this.SERVICE_URL+this.CONTROLLER_NAME;

  async auth(authname: string, password: string) : Promise<AuthResponse | null>{
    let METHOD_NAME = 'auth';
    let url = `${this.CONTROLLER_URL}/${METHOD_NAME}`;
    let headers = this.getBaseHeaders();
    let params = {};
    let body = new AuthData(authname, password);

    return await this.http.post(url, headers, params, body, false);
  }

  async register(user: User) : Promise<User | null>{
    let METHOD_NAME = 'create';
    let url = `${this.CONTROLLER_URL}/${METHOD_NAME}`;
    let headers = this.getBaseHeaders();
    let params = {};
    let body = user;

    return await this.http.post(url, headers, params, body, false);
  }
}

File: Gitbase/gitbase-desktop/Services/UsersService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using gitbase_desktop.Data;
using gitbase_desktop.Models;

namespace gitbase_desktop.Services {
    public class UsersService : AuthorizedService {
        private const string CONTROLLER_URL = SERVICE_URL + "/Users";
        public UsersService(ApplicationContext context) : base(context) { }
        public async Task<bool> Authorize(string authname, string password) {
            var url = $"{CONTROLLER_URL}/auth/authorize";
            
            var bodyData = new { Authname = authname, Password = password };
            var bodyJson = JsonSerializer.Serialize(bodyData);
            var jsonContent = new StringContent(bodyJson);
            
            var httpMessage = new HttpRequestMessage(HttpMethod.Post, url);
            httpMessage.Content = jsonContent;
            httpMessage.Headers.Add("Accept", "application/json");
            var response = await http.SendAsync(httpMessage);
            if (response.IsSuccessStatusCode) { 
                var responseText = await response.Content.ReadAsStringAsync();
                var token = JsonSerializer.Deserialize<TokenResponse>(responseText);
                context.AuthorizationToken =  new Token {
                    Self = token.Token,
                    UserId = token.User.Id
                };
                context.CurrentUser = token.User;
                return true;
            } else {
                var errorText = await response.Content.ReadAsStringAsync();
                throw new Exception(errorText);
            }
        }
        public class TokenResponse {
            public string Token { get; set; } = String.Empty;
            public User   User  { get; set; } = null;
        }
    }
}
